pkgs/native_assets_cli/test/api/build_test.dart

```diff
@@ -65,7 +65,8 @@ void main() async {
         (config, output) async {
       output.addDependency(packageRootUri.resolve('foo'));
     });
-    final buildOutputUri = outDirUri.resolve(BuildOutputImpl.fileName);
```

Comment:
It looks like this change is only because you made LinkOutput an extension type.
I don't believe it should be an extension type. Please revert this change.